federation/pkg/federation-controller/util/federated_informer.go

```diff
@@ -97,8 +100,17 @@ type FederatedInformer interface {
 // framework.DeletionHandlingMetaNamespaceKeyFunc as a keying function.
 type TargetInformerFactory func(*federation_api.Cluster, federation_release_1_4.Interface) (cache.Store, framework.ControllerInterface)
 
+type ClusterLifecycleHandler func(*federation_api.Cluster, FederationView, FederatedReadOnlyStore)
+
+type ClusterLifecycleHandlerFuncs struct {
+	ClusterAvailable   ClusterLifecycleHandler
+	ClusterUnavailable ClusterLifecycleHandler
+}
+
 // Builds a FederatedInformer for the given federation client and factory.
-func NewFederatedInformer(federationClient federation_release_1_4.Interface, targetInformerFactory TargetInformerFactory) FederatedInformer {
+func NewFederatedInformer(federationClient federation_release_1_4.Interface, targetInformerFactory TargetInformerFactory,
+	clusterLifecycle ClusterLifecycleHandlerFuncs) FederatedInformer {
```

Comment:
nit: since you are spliting this line, please split the previous one too to avoid long lines :)